File: NOVEMBER-2025/17-11-2025/1437.swift
class Solution {
    func kLengthApart(_ nums: [Int], _ k: Int) -> Bool {
        var prev = -1
        for i in 0..<nums.count {
            if nums[i] == 1 {
                if prev != -1 && (i - prev - 1) < k {
                    return false
                }
                prev = i
            }
        }
        return true
    }
}

File: NOVEMBER-2025/17-11-2025/1437.rs
impl Solution {
    pub fn k_length_apart(nums: Vec<i32>, k: i32) -> bool {
        let mut prev: i32 = -1;
        for (i, &val) in nums.iter().enumerate() {
            if val == 1 {
                if prev != -1 && (i as i32 - prev - 1) < k {
                    return false;
                }
                prev = i as i32;
            }
        }
        return true;
    }
}